### PR TITLE
U256 c fix

### PR DESCRIPTION
**Pull request by rostyslavtyshko**
## Description

There were issues in `U256` multiplication due to overflowing. Logic is fixed to take this into account

Related issue: https://github.com/FuelLabs/sway/issues/4376

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/u256_mul_test/src/main.sw [hunk 1]
```diff
@@ -33,5 +33,15 @@ fn main() -> bool {
...
 
+    let x = U256::from((0, 0, 0, 11000000000000000000));
+    let y = U256::from((0, 29, 7145508105175220139, 13399722918938673152));
+    let product = x * y;
+    let expected = U256::from((17, 9666297223066687219, 7425695065611822185, 14699749183737298944));
+    assert(product == expected);
+
+    let product_reverse = y * x;
+    let expected_reverse = U256::from((17, 9666297223066687219, 7425695065611822185, 14699749183737298944));
+    assert(product_reverse == expected_reverse);
```
Comment (Braqzen) [inline test/src/e2e_vm_tests/test_programs/should_pass/stdlib/u256_mul_test/src/main.sw:44]:
```suggestion
    let product = x * y;
    let product_reverse = y * x;
    
    let expected = U256::from((17, 9666297223066687219, 7425695065611822185, 14699749183737298944));
    
    assert(product == expected);
    assert(product_reverse == expected);
    assert(product_reverse == product);
```

**Review by Braqzen — COMMENTED** [review]
Could you link an issue, add a description etc. so that it's easier to understand what the fix is / what the code is trying to achieve?
This is a specialized area so it's difficult for me to follow all the additions / multiplications and why they are being done.

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/u256_mul_test/src/main.sw (hunk 1 above)
Comment (rostyslavtyshko) [reply test/src/e2e_vm_tests/test_programs/should_pass/stdlib/u256_mul_test/src/main.sw]: Fixed

**Review by Braqzen — APPROVED** [review]
utACK

**Review by mohammadfawaz — APPROVED** [review]

**Pull request merged**